### PR TITLE
Create ImageNet loader in keras_cv.datasets

### PR DESCRIPTION
**Pull request by ianstenbit**
@LukeWood

### CODE REVIEW

**Comment by bhack** [comment]
Why we are not using TFDS in this case?

**Comment by ianstenbit** [comment]
> Why we are not using TFDS in this case?

The preprocessed TFRecords are formatted and TPU-ready, which makes them a bit easier to plug-and-play with. TFDS ImageNet is still available as an API for those that want to use it.

@LukeWood

File: keras_cv/datasets/imagenet/load.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    # Decode label
+    label = tf.cast(tf.reshape(parsed[label_key], shape=()), dtype=tf.int32) - 1
+    label = tf.one_hot(label, 1000)
+
+    return image, label
+
+
+def load(
+    split,
+    tfrecords_path,
```
Comment (LukeWood) [inline keras_cv/datasets/imagenet/load.py:46]: nit: tfrecord_path

I think tfrecord is already plural implied (maybe I'm wrong:?)

**Review by LukeWood — CHANGES_REQUESTED** [review]
Very minor nits, thanks for putting this together

**Comment by bhack** [comment]
> > Why we are not using TFDS in this case?
> 
> The preprocessed TFRecords are formatted and TPU-ready, which makes them a bit easier to plug-and-play with. TFDS ImageNet is still available as an API for those that want to use it.
> 
> @LukeWood

Have you seen @sayakpaul gist https://gist.github.com/sayakpaul/9544d3ba935805bd47d71fd8596e7bc0#get-the-imagenet2012-data-prepared ?

**Comment by LukeWood** [comment]
Looks good!

File: keras_cv/datasets/imagenet/load.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    keys_to_features = {
+        image_key: tf.io.FixedLenFeature((), tf.string, ""),
+        label_key: tf.io.FixedLenFeature([], tf.int64, -1),
+    }
+    parsed = tf.io.parse_single_example(example, keys_to_features)
+
+    # Decode and resize image
+    image_bytes = tf.reshape(parsed[image_key], shape=[])
+    image = tf.io.decode_jpeg(image_bytes, channels=3)
+    image = layers.Resizing(
```
Comment (LukeWood) [inline keras_cv/datasets/imagenet/load.py:33]: Ah just noticed that!

constructing a Resizing every call here will cause retracing.

you need to curry a function to get around this unfortunately.

**Review by LukeWood — CHANGES_REQUESTED** [review]
Just spotted a major performance issue.  This may be bottlenecking your train script too

File: keras_cv/datasets/imagenet/load.py (hunk 2 above)
Comment (ianstenbit) [reply keras_cv/datasets/imagenet/load.py]: Ahhhh nice catch -- thanks!

**Pull request merged**